rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport.h

```diff
@@ -116,6 +116,13 @@ namespace rmw_fastrtps_cpp
 
             void deleteData(void* data);
 
+            std::function<uint32_t()> getSerializedSizeProvider(void* data) {
+              (void)data;
+              return [&]() -> uint32_t {
```

Comment:
The `&` is not there to capture `data` but to capture `this` which is necessary in order to access `members_`.